scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/daemon/DaemonEnvironment.java

```diff
@@ -48,6 +54,30 @@ public StartTask startTaskFor(ContainerInstance i) {
         .build();
   }
 
+  public List<StopTask> stopTaskFor(final List<Task> tasks) {
+    final List<String> tasksToStop =
+        tasks
+            .stream()
+            .filter(this::unmatchesTask)
+            .map(Task::getTaskDefinitionArn)
+            .collect(Collectors.toList());
+    return tasksToStop
+        .stream()
+        .map(
+            t ->
+                StopTask.builder()
+                    .clusterName(environment.getClusterName())
+                    .task(t)
+                    .reason("replaceAfterTerminate")
```

Comment:
Is this string customer-visible? If so, we should plan for making it suitably descriptive, e.g. "Stopped by deployment to EnvironmentBlah@version2".